### PR TITLE
fix dev info allocation

### PR DESCRIPTION
**Pull request by mor2code**
fix dev info allocation

### CODE REVIEW

File: cupy/cusolver.py [hunk 1]
```diff
@@ -344,7 +344,7 @@ def syevj(a, UPLO='L', with_eigen_vector=True):
 
     m, lda = a.shape
     w = _cupy.empty(m, inp_w_dtype)
-    dev_info = _cupy.empty((), _numpy.int32)
+    dev_info = _cupy.empty((1,), _cupy.int32)
```
Comment (leofang) [inline cupy/cusolver.py:347]: This change is not strictly necessary as it does allocate 4 bytes correctly.

**Comment by leofang** [comment]
btw you can modify your PR description and add "this closes #4440".

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by asi1024** [comment]
LGTM! Thanks!

**Comment by mor2code** [comment]
as far as i understand the dev_info = _cupy.empty((), _numpy.int32),
an array with memory of size zero bytes is created rather than the required
size 1 int32 of 4 bytes dev_info.data.ptr  will be incorrect without
modification

While using the batched version of syevj it exposed the problem but it same
for the non batch version which needs sizeof 1 in32

**Comment by leofang** [comment]
Hi @mor2code, I see your confusion. Note the shape `()` is not equivalent to `(0,)`: the former refers to a 0-d array (which is a scalar, backed by 4 bytes of the device memory) while the latter refers to a zero-size array (so no device memory is allocated):
```python
>>> a=cupy.empty((), cupy.int32)
>>> a
array(0, dtype=int32)
>>> a.size  # note this is not 0
1
>>> a[()]=1  # change the content
>>> a
array(1, dtype=int32)
```

**Pull request merged**